### ISSUE TITLE
checkJs on not respected with declaration files

### ISSUE THREAD

**Issue by UltraBlackLinux** [typescript-language-server/typescript-language-server#887]
Hey there,
I'm trying to use typescript declaration files to get auto completion in a project, but it's also always checking the types. The declaration files are a bit broken and don't reflect the type compatibility properly, so I'm seeing a lot of errors everywhere. I thus wanted to disable `checkJs` in the `jsconfig.json`, however the language server just continues to report errors even though it shouldn't (in my opinion anyway)
Could there be anything wrong with my setup? I'm using the default language server setup in helix, and I'm really confused why I'm running into this issue.
Any ideas?
Thanks!

**Comment by predragnikolic** [comment]
Please provide a small repo that can reproduce the behavior you are seeing. It would at least increase the chance of your issue being resolved.

**Comment by UltraBlackLinux** [comment]
oh right I had created this issue. I had another look around in my config and I think it was my fault. I used my typescript profile for both typscript and javascript. Weird placeholder file setting in the LSP config was still set to .ts and apparently that caused the issue ... ? I to this day have no idea what that does so forgive my ignorance. Changing that extension to .js seems to have solve the whole issue...
